<!DOCTYPE html>













<html class="theme-next gemini" lang>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">


  
  
    
  
  <link rel="stylesheet" href="//cdn.jsdelivr.net/gh/theme-next/theme-next-needmoreshare2@1/needsharebutton.min.css">





  <meta http-equiv="Cache-Control" content="no-transform">
  <meta http-equiv="Cache-Control" content="no-siteapp">



  <meta name="google-site-verification" content="iT9T_22nIzRSSAOvMiQs2PHGC-4cZSLIgRgeR-suXm0">



  <meta name="msvalidate.01" content="F683345C5750C40D0E5B7B0491AE2F58">





  <meta name="baidu-site-verification" content="xACIw8ifVU">



  
  
    
  
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.6/jquery.fancybox.min.css">




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Arial:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext">
  






<link rel="stylesheet" href="/lib/font-awesome/css/font-awesome.min.css?v=4.7.0">

<link rel="stylesheet" href="/css/main.css?v=7.2.0">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png?v=7.2.0">


  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png?v=7.2.0">


  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png?v=7.2.0">


  <link rel="mask-icon" href="/images/logo.svg?v=7.2.0" color="#222">







<script id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Gemini',
    version: '7.2.0',
    sidebar: {"position":"left","display":"post","offset":12,"onmobile":false,"dimmer":false},
    back2top: true,
    back2top_sidebar: true,
    fancybox: true,
    fastclick: true,
    lazyload: true,
    tabs: true,
    motion: {"enable":false,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>


  




  <meta name="description" content="1000 Genomes 简介1000 Genomes Project（缩写为1KGP）于2008年1月启动，是一项国际研究工作，旨在建立迄今为止最详细的人类遗传变异目录。科学家计划在接下来的三年内使用新开发的技术对来自不同种族群体的至少一千名匿名参与者的基因组进行测序，这些技术更快，更便宜。 2010年，该项目完成了试验阶段，在“自然”杂志的一篇出版物中对此进行了详细描述。2012年，1092个">
<meta name="keywords" content="Gene常识">
<meta property="og:type" content="article">
<meta property="og:title" content="1000 Genomes">
<meta property="og:url" content="http://univeryinli.github.io/2019/07/18/1000-genomes/index.html">
<meta property="og:site_name" content="大卫博客">
<meta property="og:description" content="1000 Genomes 简介1000 Genomes Project（缩写为1KGP）于2008年1月启动，是一项国际研究工作，旨在建立迄今为止最详细的人类遗传变异目录。科学家计划在接下来的三年内使用新开发的技术对来自不同种族群体的至少一千名匿名参与者的基因组进行测序，这些技术更快，更便宜。 2010年，该项目完成了试验阶段，在“自然”杂志的一篇出版物中对此进行了详细描述。2012年，1092个">
<meta property="og:locale" content="default">
<meta property="og:image" content="https://s2.ax1x.com/2019/07/18/ZXCeYD.jpg">
<meta property="og:updated_time" content="2019-09-09T10:36:36.946Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="1000 Genomes">
<meta name="twitter:description" content="1000 Genomes 简介1000 Genomes Project（缩写为1KGP）于2008年1月启动，是一项国际研究工作，旨在建立迄今为止最详细的人类遗传变异目录。科学家计划在接下来的三年内使用新开发的技术对来自不同种族群体的至少一千名匿名参与者的基因组进行测序，这些技术更快，更便宜。 2010年，该项目完成了试验阶段，在“自然”杂志的一篇出版物中对此进行了详细描述。2012年，1092个">
<meta name="twitter:image" content="https://s2.ax1x.com/2019/07/18/ZXCeYD.jpg">





  
  
  <link rel="canonical" href="http://univeryinli.github.io/2019/07/18/1000-genomes/">



<script id="page.configurations">
  CONFIG.page = {
    sidebar: "",
  };
</script>

  <title>1000 Genomes | 大卫博客</title>
  






  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?60ba412842b8b05e89462003493a210c";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>







  <noscript>
  <style>
  .use-motion .motion-element,
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-title { opacity: initial; }

  .use-motion .logo,
  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="default">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta">
    

    <div class="custom-logo-site-title">
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">大卫博客</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
    
      
        <h1 class="site-subtitle" itemprop="description">小流汇江海</h1>
      
    
    
  </div>

  <div class="site-nav-toggle">
    <button aria-label="Toggle navigation bar">
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>



<nav class="site-nav">
  
    <ul id="menu" class="menu">
      
        
        
        
          
          <li class="menu-item menu-item-home">

    
    
      
    

    
      
    

    <a href="/" rel="section"><i class="menu-item-icon fa fa-fw fa-home"></i> <br>Home</a>

  </li>
        
        
        
          
          <li class="menu-item menu-item-timelines">

    
    
      
    

    
      
    

    <a href="/timelines" rel="section"><i class="menu-item-icon fa fa-fw fa-history"></i> <br>Timelines</a>

  </li>
        
        
        
          
          <li class="menu-item menu-item-about">

    
    
      
    

    
      
    

    <a href="/about/" rel="section"><i class="menu-item-icon fa fa-fw fa-user"></i> <br>About</a>

  </li>
        
        
        
          
          <li class="menu-item menu-item-tags">

    
    
      
    

    
      
    

    <a href="/tags/" rel="section"><i class="menu-item-icon fa fa-fw fa-tags"></i> <br>Tags<span class="badge">18</span></a>

  </li>
        
        
        
          
          <li class="menu-item menu-item-categories">

    
    
      
    

    
      
    

    <a href="/categories/" rel="section"><i class="menu-item-icon fa fa-fw fa-th"></i> <br>Categories<span class="badge">12</span></a>

  </li>
        
        
        
          
          <li class="menu-item menu-item-archives">

    
    
      
    

    
      
    

    <a href="/archives/" rel="section"><i class="menu-item-icon fa fa-fw fa-archive"></i> <br>Archives<span class="badge">15</span></a>

  </li>

      
      
        <li class="menu-item menu-item-search">
          
            <a href="javascript:;" class="popup-trigger">
          
            
              <i class="menu-item-icon fa fa-search fa-fw"></i> <br>Search</a>
        </li>
      
    </ul>
  

  

  
    <div class="site-search">
      
  <div class="popup search-popup local-search-popup">
  <div class="local-search-header clearfix">
    <span class="search-icon">
      <i class="fa fa-search"></i>
    </span>
    <span class="popup-btn-close">
      <i class="fa fa-times-circle"></i>
    </span>
    <div class="local-search-input-wrapper">
      <input autocomplete="off" placeholder="Searching..." spellcheck="false" type="text" id="local-search-input">
    </div>
  </div>
  <div id="local-search-result"></div>
</div>



    </div>
  
</nav>



  



</div>
    </header>

    
  
  

  

  <span class="exturl github-corner" data-url="aHR0cHM6Ly9naXRodWIuY29tL3VuaXZlcnlpbmxp" title="Follow me on GitHub" aria-label="Follow me on GitHub"><svg width="80" height="80" viewbox="0 0 250 250" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"/><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"/><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"/></svg></span>



    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          
            

          
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  
    <div class="reading-progress-bar"></div>
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://univeryinli.github.io/2019/07/18/1000-genomes/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="塘朗爬坡王">
      <meta itemprop="description" content="NLP、Deep Leaning、Travel&Bicycle">
      <meta itemprop="image" content="https://s2.ax1x.com/2019/06/23/ZPsmxP.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="大卫博客">
    </span>

    
      <header class="post-header">

        
        
          <h2 class="post-title" itemprop="name headline">1000 Genomes

              
            
          </h2>
        

        <div class="post-meta">

          
          
          

          
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">Posted on</span>
              

              
                
              

              <time title="Created: 2019-07-18 13:12:15" itemprop="dateCreated datePublished" datetime="2019-07-18T13:12:15+00:00">2019-07-18</time>
            </span>
          

          
            

            
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="fa fa-calendar-check-o"></i>
                </span>
                
                  <span class="post-meta-item-text">Edited on</span>
                
                <time title="Modified: 2019-09-09 10:36:36" itemprop="dateModified" datetime="2019-09-09T10:36:36+00:00">2019-09-09</time>
              </span>
            
          

          
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">In</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing"><a href="/categories/生物信息学/" itemprop="url" rel="index"><span itemprop="name">生物信息学</span></a></span>

                
                
              
            </span>
          

          
            
            
              
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="fa fa-comment-o"></i>
                </span>
            
                <span class="post-meta-item-text">Comments: </span>
                <a href="/2019/07/18/1000-genomes/#comments" itemprop="discussionUrl">
                  <span class="post-comments-count valine-comment-count" data-xid="/2019/07/18/1000-genomes/" itemprop="commentCount"></span>
                </a>
              </span>
            
          

          
          
            <span id="/2019/07/18/1000-genomes/" class="post-meta-item leancloud_visitors" data-flag-title="1000 Genomes">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              
                <span class="post-meta-item-text">Views: </span>
              
                <span class="leancloud-visitors-count"></span>
            </span>
          

          

          <br>
          
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-file-word-o"></i>
              </span>
              
                <span class="post-meta-item-text">Symbols count in article: </span>
              
              <span title="Symbols count in article">28k</span>
            </span>
          

          
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-clock-o"></i>
              </span>
              
                <span class="post-meta-item-text">Reading time &asymp;</span>
              
              <span title="Reading time">10 mins.</span>
            </span>
          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <h1 id="1000-Genomes-简介"><a href="#1000-Genomes-简介" class="headerlink" title="1000 Genomes 简介"></a>1000 Genomes 简介</h1><p><strong>1000 Genomes Project</strong>（缩写为1KGP）于2008年1月启动，是一项国际研究工作，旨在建立迄今为止最详细的人类遗传变异目录。科学家计划在接下来的三年内使用新开发的技术对来自不同种族群体的至少一千名匿名参与者的基因组进行测序，这些技术更快，更便宜。 2010年，该项目完成了试验阶段，在“自然”杂志的一篇出版物中对此进行了详细描述。2012年，1092个基因组的测序在Nature出版物中公布。 2015年，“自然”杂志上的两篇论文报告了结果，项目的完成以及未来研究的机会。确定了许多罕见的变异，仅限于密切相关的群体，并分析了8个结构变异类别。</p>
<p>该项目将来自世界各地研究所的多学科研究团队联合起来，包括中国，意大利，日本，肯尼亚，尼日利亚，秘鲁，英国和美国。每一个都将为庞大的序列数据集和精细的人类基因组图谱做出贡献，这些图谱将通过公共数据库免费提供给科学界和公众。</p>
<p>1000 Genome Project 的目标是发现在人群中频率大于1%的变异位点，对来自不同人群的大量样本进行测序，识别到了许多的变异位点，为人类遗传变异的研究提供了一个综合的资源。</p>
<a id="more"></a>

<h2 id="基因组相关网址："><a href="#基因组相关网址：" class="headerlink" title="基因组相关网址："></a>基因组相关网址：</h2><p><span class="exturl" data-url="ZnRwOi8vZnRwLXRyYWNlLm5jYmkubmloLmdvdi8xMDAwZ2Vub21lcy9mdHAv" title="ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/">ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/<i class="fa fa-external-link"></i></span><br>[<span class="exturl" data-url="ZnRwOi8vZnRwLnNhbmdlci5hYy51ay9wdWIvMTAwMGdlbm9tZXMvXSkoZnRwOi8vZnRwLnNhbmdlci5hYy51ay9wdWIvMTAwMGdlbm9tZXMvKQ==" title="ftp://ftp.sanger.ac.uk/pub/1000genomes/])(ftp://ftp.sanger.ac.uk/pub/1000genomes/)">ftp://ftp.sanger.ac.uk/pub/1000genomes/])(ftp://ftp.sanger.ac.uk/pub/1000genomes/)<i class="fa fa-external-link"></i></span><br><span class="exturl" data-url="ZnRwOi8vZnRwLmViaS5hYy51ay9wdWIvZGF0YWJhc2VzLzEwMDBnZW5vbWVz" title="ftp://ftp.ebi.ac.uk/pub/databases/1000genomes">ftp://ftp.ebi.ac.uk/pub/databases/1000genomes<i class="fa fa-external-link"></i></span><br><span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cA==" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp<i class="fa fa-external-link"></i></span></p>
<p>uk 10k gene:<br><span class="exturl" data-url="aHR0cHM6Ly93d3cudWsxMGsub3JnL2RhdGFfYWNjZXNzLmh0bWw=" title="https://www.uk10k.org/data_access.html">https://www.uk10k.org/data_access.html<i class="fa fa-external-link"></i></span></p>
<p>生信人必学ftp站点之1000genomes,总结了很多可用的站点：<br><span class="exturl" data-url="aHR0cDovL3d3dy5iaW8taW5mby10cmFpbmVlLmNvbS8xODQxLmh0bWw=" title="http://www.bio-info-trainee.com/1841.html">http://www.bio-info-trainee.com/1841.html<i class="fa fa-external-link"></i></span></p>
<p>vcf文件格式说明：<br><span class="exturl" data-url="aHR0cDovL3d3dy5vbWljc2NsYXNzLmNvbS9hcnRpY2xlLzY=" title="http://www.omicsclass.com/article/6">http://www.omicsclass.com/article/6<i class="fa fa-external-link"></i></span></p>
<h2 id="基因组计划详细说明"><a href="#基因组计划详细说明" class="headerlink" title="基因组计划详细说明"></a>基因组计划详细说明</h2><p>人类基因组由大约30亿个DNA碱基对组成，估计携带约20,000个蛋白质编码基因。在设计研究时，该联盟需要解决有关项目指标的若干关键问题，如技术挑战，数据质量标准和序列覆盖。</p>
<p>整个项目划分为四个阶段，<strong>试点阶段</strong>和三个<strong>主要阶段</strong>。</p>
<ol>
<li>为了确定整个项目的最终设计，设计了三个试点研究，并将在项目的第一年内进行<ul>
<li>第一个试点旨在对低覆盖率（2x）的3个主要地理群体的180个人进行基因分型。</li>
<li>第二项初步研究，两个核心家族（父母和成年子女）的基因组将进行深度覆盖（每个基因组20倍）的测序。</li>
<li>第三项试点研究涉及对1000名深度覆盖（20x）的1000个基因的编码区（外显子）进行测序。</li>
</ul>
</li>
<li>主要阶段中只有第一阶段和第三阶段产生了数据，结果发现，平均而言，每个人在注释基因中携带约250-300个功能丧失变体，并且先前涉及遗传性疾病的50-100个变体。<ul>
<li>来自4个群体的180个个体的低覆盖度全基因组测序</li>
<li>2个三人组（母亲 - 孩子）的高覆盖率排序</li>
<li>来自7个群体的697个个体的外显子靶向测序</li>
</ul>
</li>
</ol>
<p>整个项目从2008年开始到2013年结束，最终的版本为2013年5月2日发布的数据, 包含了来自26个人群，共2504个样本的SNP分型结果。根据Fort Lauderdale principles原则，所有基因组序列数据（包括变体调用）随着项目的进展免费提供，1000G的数据是免费公开的，可以通过ftp下载得到。</p>
<h1 id="1000-Genomes-数据库结构-ftp"><a href="#1000-Genomes-数据库结构-ftp" class="headerlink" title="1000 Genomes 数据库结构-ftp"></a>1000 Genomes 数据库结构-ftp</h1><p>数据存储在ftp上，其目录结构如下图，包含多个说明文件*.md：</p>
<ul>
<li><p>README_ebi_aspera_info.md–高速下载说明</p>
</li>
<li><p>README_file_formats_and_descriptions.md–文件格式和描述</p>
</li>
<li><p>README_ftp_site_structure.md–ftp网站文件结构</p>
</li>
<li><p>README_missing_files.md–缺失文件</p>
</li>
<li><p>README_populations.md–人群</p>
</li>
<li><p>README_using_1000genomes_cram.md–cram文件读取</p>
</li>
<li><p>data_collections–数据存储的文件夹</p>
</li>
</ul>
<p><img src="https://s2.ax1x.com/2019/07/18/ZXCeYD.jpg" alt="ZXCeYD.jpg"></p>
<p>README_ebi_aspera_info.md–高速下载说明</p>
<blockquote>
<h2 id="How-to-download-files-using-Aspera"><a href="#How-to-download-files-using-Aspera" class="headerlink" title="How to download files using Aspera"></a>How to download files using Aspera</h2><h3 id="Download-Aspera"><a href="#Download-Aspera" class="headerlink" title="Download Aspera"></a>Download Aspera</h3><p>Aspera provides a fast method of downloading data. To use the Aspera service you need to download the <span class="exturl" data-url="aHR0cDovL2FzcGVyYXNvZnQuY29tL3NvZnR3YXJlL3RyYW5zZmVyLWNsaWVudHMvY29ubmVjdC13ZWItYnJvd3Nlci1wbHVnLWluLw==" title="http://asperasoft.com/software/transfer-clients/connect-web-browser-plug-in/">Aspera connect<i class="fa fa-external-link"></i></span> software. This provides a bulk download client called ascp.</p>
<h3 id="Browser"><a href="#Browser" class="headerlink" title="Browser"></a>Browser</h3><p>Our aspera browser interace no longer works. If you wish to download files using a web interface we recommend using the <a href="/faq/can-i-access-1000-genomes-data-globus-online/">Globus interface we present</a>. If you are previously relied on the aspera web interface and wish to discuss the matter please email us at <span class="exturl" data-url="bWFpbHRvOmluZm9AMTAwMGdlbm9tZXMub3Jn" title="mailto:info@1000genomes.org">info@1000genomes.org<i class="fa fa-external-link"></i></span> to discuss your options. </p>
<h3 id="Command-line"><a href="#Command-line" class="headerlink" title="Command line"></a>Command line</h3><p>For the command line tool ascp, for versions 3.3.3 and newer, you need to use a command line like:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt;   ascp -i bin/aspera/etc/asperaweb_id_dsa.openssh -Tr -Q -l 100M -P33001 -L- fasp-g1k@fasp.1000genomes.ebi.ac.uk:vol1/ftp/release/20100804/ALL.2of4intersection.20100804.genotypes.vcf.gz ./</span><br><span class="line">&gt;</span><br></pre></td></tr></table></figure>
</blockquote>
<blockquote>
<p>For versions 3.3.2 and older, you need to use a command line like:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt;   ascp -i bin/aspera/etc/asperaweb_id_dsa.putty -Tr -Q -l 100M -P33001 -L- fasp-g1k@fasp.1000genomes.ebi.ac.uk:vol1/ftp/release/20100804/ALL.2of4intersection.20100804.genotypes.vcf.gz ./</span><br><span class="line">&gt;</span><br></pre></td></tr></table></figure>
</blockquote>
<blockquote>
<p>Note, the only change between these commands is that for newer versions of ascp asperaweb_id_dsa.openssh replaces asperaweb_id_dsa.putty. This change is noted by Aspera <span class="exturl" data-url="aHR0cHM6Ly9zdXBwb3J0LmFzcGVyYXNvZnQuY29tL2VudHJpZXMvMzg2NzU0NjgtQ29tbWFuZC1saW5lLWFzY3AtdHJhbnNmZXItYXNraW5nLWZvci1hLXBhc3NwaHJhc2UtYWZ0ZXItQ29ubmVjdC1wbHVnaW4tdXBncmFkZQ==" title="https://support.asperasoft.com/entries/38675468-Command-line-ascp-transfer-asking-for-a-passphrase-after-Connect-plugin-upgrade">here<i class="fa fa-external-link"></i></span>. You can check the version of ascp you have using:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt; ascp --version</span><br><span class="line">&gt;</span><br></pre></td></tr></table></figure>
</blockquote>
<blockquote>
<p>The argument to -i may also be different depending on the location of the default key file. The command should not ask you for a password. All the IGSR data is accessible without a password but you do need to give ascp the ssh key to complete the command.</p>
<p>For the above commands to work with your network’s firewall you need to open ports 22/tcp (outgoing) and 33001/udp (both incoming and outgoing) to the following EBI IPs:</p>
<ul>
<li>193.62.192.6</li>
<li>193.62.193.6</li>
<li>193.62.193.135</li>
</ul>
<p>If the firewall has UDP flood protection, it must be turned off for port 33001.</p>
<h3 id="Further-details"><a href="#Further-details" class="headerlink" title="Further details"></a>Further details</h3><p>For further information, please contact <span class="exturl" data-url="bWFpbHRvOmluZm9AMTAwMGdlbm9tZXMub3Jn" title="mailto:info@1000genomes.org">info@1000genomes.org<i class="fa fa-external-link"></i></span>.</p>
</blockquote>
<p>README_file_formats_and_descriptions.md–文件格式和描述</p>
<blockquote>
<h2 id="File-formats-and-descriptions"><a href="#File-formats-and-descriptions" class="headerlink" title="File formats and descriptions"></a>File formats and descriptions</h2><p>This file provides information on some of the file formats used to make data available on this site.</p>
<h3 id="CRAM"><a href="#CRAM" class="headerlink" title="CRAM"></a>CRAM</h3><p>.cram files use a reference-based format to store data. This format is being used to supply alignment data on this site. Detailed information on working with our .cram files is provided in <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9SRUFETUVfdXNpbmdfMTAwMGdlbm9tZXNfY3JhbS5tZA==" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/README_using_1000genomes_cram.md">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/README_using_1000genomes_cram.md<i class="fa fa-external-link"></i></span>.</p>
<h3 id="CRAI"><a href="#CRAI" class="headerlink" title="CRAI"></a>CRAI</h3><p>.crai files are index files which accompany CRAM files. They must be present in order to work with .cram files for most purposes.</p>
<h3 id="BAS"><a href="#BAS" class="headerlink" title="BAS"></a>BAS</h3><p>.bas files are .cram or .bam *<em>statistic files *</em>with one line per readgroup and columns separated by<br>tabs. The first line is a header that describes each column. The first six columns<br>provide meta information about each readgroup.</p>
<p>The remaining columns provide various statistics about the readgroup, calculated<br>by going through the release bams. Where data isn’t available to calculate the<br>result for a column, the default value will be 0.</p>
<p>Each column is described in detail below:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">&gt;       Column 1  &apos;bam_filename&apos;: The DCC bam file name in which the readgroup data can be found.</span><br><span class="line">&gt;       Column 2  &apos;md5&apos;: The md5 checksum of the bam file named in column 1.</span><br><span class="line">&gt;       Column 3  &apos;study&apos;: The SRA study id this readgroup belongs to.</span><br><span class="line">&gt;       Column 4  &apos;sample&apos;: The sample (individual) identifier the readgroup came from.</span><br><span class="line">&gt;       Column 5  &apos;platform&apos;: The sequencing platform (technology) used to sequence the readgroup.</span><br><span class="line">&gt;       Column 6  &apos;library&apos;: The name of the library used for the readgroup.</span><br><span class="line">&gt;       Column 7  &apos;readgroup&apos;: The readgroup identifier. This is unique per .bas file. The remaining columns summarise data for reads with this RG tag in the bam file given in column 1.</span><br><span class="line">&gt;       Column 8  &apos;#_total_bases&apos;: The sum of the length of all reads in this readgroup.</span><br><span class="line">&gt;       Column 9  &apos;#_mapped_bases&apos;: The sum of the length of all reads in this readgroup that did not have flag 4 (== unmapped).</span><br><span class="line">&gt;       Column 10 &apos;#_total_reads&apos;: The total number of reads in this readgroup.</span><br><span class="line">&gt;       Column 11 &apos;#_mapped_reads&apos;: The total number of reads in this readgroup that did not have flag 4 (== unmapped).</span><br><span class="line">&gt;       Column 12 &apos;#_mapped_reads_paired_in_sequencing&apos;: As for column 10, but also requiring flag 1 (== reads paired in sequecing).</span><br><span class="line">&gt;       Column 13 &apos;#_mapped_reads_properly_paired&apos;: As for column 10, but also requiring flag 2 (== mapped in a proper pair, inferred during alignment).</span><br><span class="line">&gt;       Column 14 &apos;%_of_mismatched_bases&apos;: Calculated by summing the read lengths of all reads in this readgroup that have an NM tag, summing the edit distances obtained from the NM tags, and getting the percentage of the latter out of the former to 2 decimal places.</span><br><span class="line">&gt;       Column 15 &apos;average_quality_of_mapped_bases&apos;: The mean of all the base qualities of the bases counted for column 8, to 2 decimal places.</span><br><span class="line">&gt;       Column 16 &apos;mean_insert_size&apos;: The mean of all insert sizes (ISIZE field) greater than 0 for properly paired reads (as counted in column 12) and with a mapping quality (MAPQ field) greater than 0. Rounded to the nearest whole number.</span><br><span class="line">&gt;       Column 17 &apos;insert_size_sd&apos;: The standard deviation from the mean of insert sizes considered for column 15. To 2 decimal places.</span><br><span class="line">&gt;       Column 18 &apos;median_insert_size&apos;: The median insert size, using the same set of insert sizes considered for column 15.</span><br><span class="line">&gt;       Column 19 &apos;insert_size_median_absolute_deviation&apos;: The median absolute deviation of the column 17 data.</span><br><span class="line">&gt;       Column 20 &apos;#_duplicate_reads&apos;: The number of reads which were marked as duplicates.</span><br><span class="line">&gt;       Column 21 &apos;#_duplicate_bases&apos;: The number of bases which were narked as duplicated</span><br><span class="line">&gt;</span><br></pre></td></tr></table></figure>
</blockquote>
<blockquote>
<h3 id="INDEX"><a href="#INDEX" class="headerlink" title="INDEX"></a>INDEX</h3><p>Various types of index file exist on the site. These are tab-delimited files where the data is arranged in columns. Immediately before the body of the file there is a header line, which starts with #, that gives the column names. In addition, index files may have further information at the head of the file. These lines start with ## and can provide descriptions of the columns, the date the index was generated and other pieces of information, as appropriate to the file and data set.</p>
<p>An example of the start of such a file, in this case an alignment index file, is below:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">&gt;   ##FileDate=20150914</span><br><span class="line">&gt;   ##Project=Illumina Platinum pedigree</span><br><span class="line">&gt;   ##CRAM_FILE=Path to CRAM file - information on CRAM files can be found in ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/README_using_1000genomes_cram.md and information on the alignment process accompanies the data collection</span><br><span class="line">&gt;   ##CRAM_MD5=md5 for CRAM file</span><br><span class="line">&gt;   ##CRAI_FILE=Path to CRAI file</span><br><span class="line">&gt;   ##CRAI_MD5=md5 for CRAI file</span><br><span class="line">&gt;   ##BAS_FILE=Path to BAS file</span><br><span class="line">&gt;   ##BAS_MD5=md5 for BAS file</span><br><span class="line">&gt;   #CRAM_FILE	CRAM_MD5	CRAI_FILE	CRAI_MD5	BAS_FILE	BAS_MD5</span><br><span class="line">&gt;   ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/illumina_platinum_pedigree/data/CEU/NA12893/alignment/NA12893.alt_bwamem_GRCh38DH.20150706.CEU.illumina_platinum_ped.cram	4aa7f5b61d4365a556c980278b9be5a1	ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/illumina_platinum_pedigree/data/CEU/NA12893/alignment/NA12893.alt_bwamem_GRCh38DH.20150706.CEU.illumina_platinum_ped.cram.crai	ae3d1ac0de67d58d192a96508bad85b4	ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/illumina_platinum_pedigree/data/CEU/NA12893/alignment/NA12893.alt_bwamem_GRCh38DH.20150706.CEU.illumina_platinum_ped.bam.bas	7fb61b29c6b5fc716e9167affab56d92</span><br><span class="line">&gt;   ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/illumina_platinum_pedigree/data/CEU/NA12892/alignment/NA12892.alt_bwamem_GRCh38DH.20150706.CEU.illumina_platinum_ped.cram	8acdcd17349546b5ca1b45111e30fc07	ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/illumina_platinum_pedigree/data/CEU/NA12892/alignment/NA12892.alt_bwamem_GRCh38DH.20150706.CEU.illumina_platinum_ped.cram.crai	555de96d6baf2e16630dadd9a6b5b038	ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/illumina_platinum_pedigree/data/CEU/NA12892/alignment/NA12892.alt_bwamem_GRCh38DH.20150706.CEU.illumina_platinum_ped.bam.bas	5079d6d9dfcb0eb4631d11035ec71b16</span><br><span class="line">&gt;</span><br></pre></td></tr></table></figure>
</blockquote>
<blockquote>
<h3 id="Further-information"><a href="#Further-information" class="headerlink" title="Further information"></a>Further information</h3><p>For further information, please contact <span class="exturl" data-url="bWFpbHRvOmluZm9AMTAwMGdlbm9tZXMub3Jn" title="mailto:info@1000genomes.org">info@1000genomes.org<i class="fa fa-external-link"></i></span>.</p>
</blockquote>
<p>README_ftp_site_structure.md–ftp网站文件结构</p>
<blockquote>
<h2 id="Structure-of-the-FTP-site"><a href="#Structure-of-the-FTP-site" class="headerlink" title="Structure of the FTP site"></a>Structure of the FTP site</h2><p>This file provides an overview of the structure of the FTP site.</p>
<h3 id="Top-level-of-the-site"><a href="#Top-level-of-the-site" class="headerlink" title="Top level of the site"></a>Top level of the site</h3><p>At the top level of the site, <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC8=" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/<i class="fa fa-external-link"></i></span>, there are a number of files and directories.</p>
<p>Files present in <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC8=" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/<i class="fa fa-external-link"></i></span>:</p>
<ul>
<li>README files: These provide information on a variety of topics related to this FTP site and the data it contains.</li>
<li>CHANGELOG file: This file records alterations made to this FTP site.</li>
<li>current.tree file: This file lists all directories and files currently present on this FTP site.</li>
</ul>
<p>Directories present in <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC8=" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/<i class="fa fa-external-link"></i></span>:</p>
<ul>
<li>changelog_details</li>
<li>data</li>
<li>data_collections</li>
<li>historical_data</li>
<li>phase1</li>
<li>phase3</li>
<li>pilot_data</li>
<li>release</li>
<li>technical</li>
</ul>
<p>These directories and described in the next section of this file.</p>
<h3 id="Directories"><a href="#Directories" class="headerlink" title="Directories"></a>Directories</h3><h4 id="changelog-details"><a href="#changelog-details" class="headerlink" title="changelog_details"></a>changelog_details</h4><p>This directory contains a series of files detailing the changes made to the FTP site over time.</p>
<h4 id="data"><a href="#data" class="headerlink" title="data"></a>data</h4><p>The data directory formerly housed data generated during the 1000 Genomes Project. The data that was previously located here has been integrated into data_collections and is present under <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9kYXRhX2NvbGxlY3Rpb25zLzEwMDBfZ2Vub21lc19wcm9qZWN0L2RhdGE=" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/data">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/data<i class="fa fa-external-link"></i></span>. Further information on this move can be found in <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9kYXRhL1JFQURNRV9kYXRhX2hhc19tb3ZlZC5tZA==" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data/README_data_has_moved.md">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data/README_data_has_moved.md<i class="fa fa-external-link"></i></span>.</p>
<h4 id="data-collections"><a href="#data-collections" class="headerlink" title="data_collections"></a>data_collections</h4><p>The data_collections directory contains directories for various collections of data, typically generated by different projects. Among the data collections is the <strong>1000 Genomes Project</strong> data.</p>
<p>For each collection of data, within the directory for that collection, README and index files provide information on the collection. Under each collection directory, there is a data directory, under which files are organised by population and then sample. Further information can be found in <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9kYXRhX2NvbGxlY3Rpb25zL1JFQURNRV9kYXRhX2NvbGxlY3Rpb25zLm1k" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/README_data_collections.md">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/README_data_collections.md<i class="fa fa-external-link"></i></span>.</p>
<h4 id="historical-data"><a href="#historical-data" class="headerlink" title="historical_data"></a>historical_data</h4><p>This directory was created during a rearrangement of the FTP site in September 2015. It houses README and index files that were formerly present at the toplevel of this site, including dedicated index directories. Further information is available in <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9oaXN0b3JpY2FsX2RhdGEvUkVBRE1FX2hpc3RvcmljYWxfZGF0YS5tZA==" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/historical_data/README_historical_data.md">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/historical_data/README_historical_data.md<i class="fa fa-external-link"></i></span>.</p>
<h4 id="phase1"><a href="#phase1" class="headerlink" title="phase1"></a>phase1</h4><p>This directory contains data that supports the publications associated with phase 1 of the 1000 Genomes Project.</p>
<h4 id="phase3"><a href="#phase3" class="headerlink" title="phase3"></a>phase3</h4><p>This directory contains data that supports the publications associated with phase 3 of the 1000 Genomes Project.</p>
<h4 id="pilot-data"><a href="#pilot-data" class="headerlink" title="pilot_data"></a>pilot_data</h4><p>This directory contains data that supports the publications associated with the pilot phase of the 1000 Genomes Project.</p>
<h4 id="release"><a href="#release" class="headerlink" title="release"></a>release</h4><p>The release directory contains dated directories which contain analysis results sets plus README files explaining how those data sets were produced.</p>
<p>Originally, the date in release subdirectory names was the date on which the given release was made. Thereafter, the release subdirectory dates were based on the date in the name of the corresponding YYYYMMDD.sequence.index file. In future, the date in the directory name will be chosen in a manner appropriate to the data and the nature of the release.</p>
<p>Examples of release subdirectories are: </p>
<ul>
<li><span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay9yZWxlYXNlLzIwMDhfMTIv" title="ftp://ftp.1000genomes.ebi.ac.uk/release/2008_12/">ftp://ftp.1000genomes.ebi.ac.uk/release/2008_12/<i class="fa fa-external-link"></i></span></li>
<li><span class="exturl" data-url="ZnRwOi8vZnRwLXRyYWNlLm5jYmkubmloLmdvdi8xMDAwZ2Vub21lcy9yZWxlYXNlLzIwMDhfMTIv" title="ftp://ftp-trace.ncbi.nih.gov/1000genomes/release/2008_12/">ftp://ftp-trace.ncbi.nih.gov/1000genomes/release/2008_12/<i class="fa fa-external-link"></i></span></li>
</ul>
<p>In cases where release directories are named based on the date of the YYYYMMDD.sequence.index, the SNP calls, indel calls, etc. in these directories are based on alignments produced from data listed in the  YYYYMMDD.sequence.index file.</p>
<p>For example, the directory<br><span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9yZWxlYXNlLzIwMTAwODA0Lw==" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20100804/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20100804/<i class="fa fa-external-link"></i></span><br>contains the release versions of SNP and indel calls based on the<br><span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9oaXN0b3JpY2FsX2RhdGEvZm9ybWVyX3RvcGxldmVsL3NlcXVlbmNlX2luZGljZXMvMjAxMDA4MDQuc2VxdWVuY2UuaW5kZXg=" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/historical_data/former_toplevel/sequence_indices/20100804.sequence.index">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/historical_data/former_toplevel/sequence_indices/20100804.sequence.index<i class="fa fa-external-link"></i></span><br>file.</p>
<h4 id="technical"><a href="#technical" class="headerlink" title="technical"></a>technical</h4><p>The technical directory contains subdirectories for other data sets such as simulations, files for<br>method development, interm data sets, reference genomes, etc..</p>
<p>An example of data stored under technical is <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay90ZWNobmljYWwvc2ltdWxhdGlvbnMv" title="ftp://ftp.1000genomes.ebi.ac.uk/technical/simulations/">ftp://ftp.1000genomes.ebi.ac.uk/technical/simulations/<i class="fa fa-external-link"></i></span>.</p>
<p><strong><em>WARNING: <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay90ZWNobmljYWwvd29ya2luZy8=" title="ftp://ftp.1000genomes.ebi.ac.uk/technical/working/">ftp://ftp.1000genomes.ebi.ac.uk/technical/working/<i class="fa fa-external-link"></i></span></em></strong></p>
<p>The working directory under technical contains data that has experimental (non-public release) status<br>and is suitable for internal project use only. Please use with <strong><em>caution</em></strong>. </p>
<h3 id="Further-information-1"><a href="#Further-information-1" class="headerlink" title="Further information"></a>Further information</h3><p>Should you require further assistance in navigating the FTP site, please contact <span class="exturl" data-url="bWFpbHRvOmluZm9AMTAwMGdlbm9tZXMub3Jn" title="mailto:info@1000genomes.org">info@1000genomes.org<i class="fa fa-external-link"></i></span>.</p>
</blockquote>
<p>README_missing_files.md–缺失文件</p>
<blockquote>
<p>The 1000 Genomes FTP site under goes periodic rearrangments to accomodate new data sets coming in<br>and to preserve old data in a new location.</p>
<p>If there is a particular file where you don’t know if it still exists, the best place to try is<br>our current.tree file.</p>
<p><span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9jdXJyZW50LnRyZWU=" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/current.tree">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/current.tree<i class="fa fa-external-link"></i></span></p>
<p>This file contains five columns:</p>
<ul>
<li>relative path</li>
<li>type (directory or file)</li>
<li>size in bytes</li>
<li>last updated time stamp</li>
<li>md5</li>
</ul>
<p>You should be able to search this file with your filename and see if the file still exists and find its new location.</p>
<p>If you still can’t find your file email <span class="exturl" data-url="bWFpbHRvOmluZm9AMTAwMGdlbm9tZXMub3Jn" title="mailto:info@1000genomes.org">info@1000genomes.org<i class="fa fa-external-link"></i></span> and the support team will be able to tell you what has<br>happened to it.</p>
</blockquote>
<p>README_populations.md–人群</p>
<blockquote>
<h2 id="Populations"><a href="#Populations" class="headerlink" title="Populations"></a>Populations</h2><p>This file describes the population codes where assigned to samples collected for the 1000 Genomes project. These codes are used to organise the files in the data_collections’ project data directories and can also be found in column 11 of many sequence index files.</p>
<p>There are also two tsv files, which contain the population codes and descriptions for both the sub and super populations that were used in phase 3 of the 1000 Genomes Project:</p>
<p><span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9waGFzZTMvMjAxMzEyMTkucG9wdWxhdGlvbnMudHN2" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/20131219.populations.tsv">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/20131219.populations.tsv<i class="fa fa-external-link"></i></span><br><span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9waGFzZTMvMjAxMzEyMTkuc3VwZXJwb3B1bGF0aW9ucy50c3Y=" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/20131219.superpopulations.tsv">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/20131219.superpopulations.tsv<i class="fa fa-external-link"></i></span></p>
<h3 id="Populations-and-codes"><a href="#Populations-and-codes" class="headerlink" title="Populations and codes"></a>Populations and codes</h3><pre><code>CHB    Han Chinese             Han Chinese in Beijing, China
JPT    Japanese                Japanese in Tokyo, Japan
CHS    Southern Han Chinese    Han Chinese South
CDX    Dai Chinese             Chinese Dai in Xishuangbanna, China
KHV    Kinh Vietnamese         Kinh in Ho Chi Minh City, Vietnam
CHD    Denver Chinese          Chinese in Denver, Colorado (pilot 3 only)

CEU    CEPH                    Utah residents (CEPH) with Northern and Western European ancestry 
TSI    Tuscan                  Toscani in Italia 
GBR    British                 British in England and Scotland 
FIN    Finnish                 Finnish in Finland 
IBS    Spanish                 Iberian populations in Spain 

YRI    Yoruba                  Yoruba in Ibadan, Nigeria
LWK    Luhya                   Luhya in Webuye, Kenya
GWD    Gambian                 Gambian in Western Division, The Gambia 
MSL    Mende                   Mende in Sierra Leone
ESN    Esan                    Esan in Nigeria

ASW    African-American SW     African Ancestry in Southwest US  
ACB    African-Caribbean       African Caribbean in Barbados
MXL    Mexican-American        Mexican Ancestry in Los Angeles, California
PUR    Puerto Rican            Puerto Rican in Puerto Rico
CLM    Colombian               Colombian in Medellin, Colombia
PEL    Peruvian                Peruvian in Lima, Peru

GIH    Gujarati                Gujarati Indian in Houston, TX
PJL    Punjabi                 Punjabi in Lahore, Pakistan
BEB    Bengali                 Bengali in Bangladesh
STU    Sri Lankan              Sri Lankan Tamil in the UK
ITU    Indian                  Indian Telugu in the UK</code></pre><p>Should you have any queries, please contact <span class="exturl" data-url="bWFpbHRvOmluZm9AMTAwMGdlbm9tZXMub3Jn" title="mailto:info@1000genomes.org">info@1000genomes.org<i class="fa fa-external-link"></i></span>.</p>
</blockquote>
<p>README_using_1000genomes_cram.md–cram文件读取</p>
<blockquote>
<h2 id="IGSR-CRAM-Tutorial"><a href="#IGSR-CRAM-Tutorial" class="headerlink" title="IGSR CRAM Tutorial"></a>IGSR CRAM Tutorial</h2><p>From the first release of GRCh38 alignments onwards, we are releasing our alignment files in <strong>CRAM</strong> format. CRAM is a reference-based compression of sequence data. </p>
<p>Both htslib and picard can read CRAM files, many standard tools should be able to read these files natively. </p>
<p>Here are details about how to view CRAM files, convert from CRAM to BAM, how we produced the cram files and the CRAM specification.</p>
<h3 id="Using-CRAM-files"><a href="#Using-CRAM-files" class="headerlink" title="Using CRAM files"></a>Using CRAM files</h3><p>CRAM Files can be read by both samtools and picard. EMBL-EBI also provides a java API called cramtools (<span class="exturl" data-url="aHR0cDovL3d3dy5lYmkuYWMudWsvZW5hL3NvZnR3YXJlL2NyYW0tdG9vbGtpdA==" title="http://www.ebi.ac.uk/ena/software/cram-toolkit">http://www.ebi.ac.uk/ena/software/cram-toolkit<i class="fa fa-external-link"></i></span>)</p>
<ul>
<li>Reading a CRAM file with samtools - samtools view commands work with CRAM files. This functionality needs samtools v1.2 or higher</li>
</ul>
<p><code>&gt;samtools view $input.cram -h chr22:1000000-1500000 | less</code></p>
<ul>
<li>Converting a CRAM file to a BAM File - some tools still need BAM files rather than CRAM. You can convert from CRAM to BAM easily</li>
</ul>
<p><code>&gt;java -jar cramtools-3.0.jar bam  -I $input.cram -R $reference.fa -O $output.bam</code></p>
<p>Please note the first time you run these commands the program reading the CRAM file must download the reference sequence data from an online cache. This process can be speeded up if you download the required reference file and build a local copy of the cache in advance. This process is described below in the CRAM reference registry section.</p>
<h3 id="The-CRAM-reference-registry"><a href="#The-CRAM-reference-registry" class="headerlink" title="The CRAM reference registry"></a>The CRAM reference registry</h3><p>Because CRAM does not contain the same level of sequence data as BAM files, it relies on the CRAM reference registry to provide reference sequences for CRAM to output uncompressed sequences.  The reference must be available at all times. Losing it is equivalent to losing all your read sequences. Retrieval of reference data from the registry is supported by using MD5 or SHA1 checksums using the following URLs:</p>
<p><code>www.ebi.ac.uk/ena/cram/md5/&lt;hashvalue&gt;</code><br><code>www.ebi.ac.uk/ena/cram/sha1/&lt;hashvalue&gt;</code></p>
<p>The md5 values for all GRCh38 reference chromosomes and contigs are included in CRAM file headers. These are mandatory fields in the CRAM specification.  </p>
<p>The following process can be used to download and prepare the cache in advance to speed up initial reads of the sequence data from CRAM files. </p>
<ol>
<li>Download the reference file from our FTP site. <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC90ZWNobmljYWwvcmVmZXJlbmNlL0dSQ2gzOF9yZWZlcmVuY2VfZ2Vub21lL0dSQ2gzOF9mdWxsX2FuYWx5c2lzX3NldF9wbHVzX2RlY295X2hsYS5mYQ==" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/GRCh38_reference_genome/GRCh38_full_analysis_set_plus_decoy_hla.fa">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/GRCh38_reference_genome/GRCh38_full_analysis_set_plus_decoy_hla.fa<i class="fa fa-external-link"></i></span></li>
<li>Run the seq_cache_populate.pl script from samtools - <span class="exturl" data-url="aHR0cDovL3d3dy5odHNsaWIub3JnL3dvcmtmbG93LyNtYXBwaW5nX3RvX2NyYW0=" title="http://www.htslib.org/workflow/#mapping_to_cram">http://www.htslib.org/workflow/#mapping_to_cram<i class="fa fa-external-link"></i></span><br><code>&gt;perl samtools/misc/seq_cache_populate.pl -root /path/to/cache /path/to/GRCh38_full_analysis_set_plus_decoy_hla.fa</code></li>
<li>Set the cache environment variables<br><code>&gt;export REF_PATH=/path/to/cache/%2s/%2s/%s:http://www.ebi.ac.uk/ena/cram/md5/%s</code><br><code>&gt;export REF_CACHE=/path/to/cache/%2s/%2s/%s</code></li>
</ol>
<h3 id="How-did-we-compress-the-BAMs-to-CRAMs"><a href="#How-did-we-compress-the-BAMs-to-CRAMs" class="headerlink" title="How did we compress the BAMs to CRAMs"></a>How did we compress the BAMs to CRAMs</h3><p>We use cramtools to convert the BAM files our alignment pipeline produces to CRAM Files. The files are compressed using a lossy mode, this bins all quality stores into the 8-binning scheme defined by Illumina.</p>
<p><code>&gt;java -jar cramtools-3.0.jar cram --ignore-tags OQ:CQ:BQ --capture-all-tags --lossy-quality-score-spec &#39;*8&#39; --preserve-read-names -O $output.cram -R GRCh38_full_analysis_set_plus_decoy_hla.fa -I $input.bam</code></p>
<h3 id="More-information-about-CRAM-format"><a href="#More-information-about-CRAM-format" class="headerlink" title="More information about CRAM format."></a>More information about CRAM format.</h3><p>As mentioned above CRAM represents a reference-based compression of sequence data. After aligning sequences reads to a reference genome, rather than storing every base pair of a sequence read, the approach stores only the difference between the read and the reference, hence reducing the space needed for storing sequence reads.  Additional compression can be archived in lossy mode by controlled loss of quality information and unaligned reads, by dropping read names and other information. The level of compression can be fine tuned based on users’ experiment design. With associated tools, the compressed data can be seamlessly uncompressed and fed into downstream analysis.</p>
<p>This compression method was first developed by Ewan Birney’s group in European Bioinformatics Institute (EBI) (Hsi-Yang Fritz, et al. (2011). Genome Res. 21:734-740).  The specification itself is maintained by the HTSlib group alongside the BAM, VCF and BCF specifications <span class="exturl" data-url="aHR0cDovL3NhbXRvb2xzLmdpdGh1Yi5pby9odHMtc3BlY3MvQ1JBTXYzLnBkZg==" title="http://samtools.github.io/hts-specs/CRAMv3.pdf">http://samtools.github.io/hts-specs/CRAMv3.pdf<i class="fa fa-external-link"></i></span>.</p>
<p>If you have any questions about our CRAM files or our alignment pipeline, please email <span class="exturl" data-url="bWFpbHRvOmluZm9AMTAwMGdlbm9tZXMub3Jn" title="mailto:info@1000genomes.org">info@1000genomes.org<i class="fa fa-external-link"></i></span></p>
</blockquote>
<h1 id="1000-Genomes-数据库结构-data-collections"><a href="#1000-Genomes-数据库结构-data-collections" class="headerlink" title="1000 Genomes 数据库结构-data_collections"></a>1000 Genomes 数据库结构-data_collections</h1><p>该文件主要存储基因组的2000人的数据，本层数据说明文件如下：</p>
<blockquote>
<h2 id="Data-collections"><a href="#Data-collections" class="headerlink" title="Data collections"></a>Data collections</h2><h3 id="Collections"><a href="#Collections" class="headerlink" title="Collections"></a>Collections</h3><p>The International Genome Sample Resource (IGSR) provides access to data from <strong>multiple</strong> projects, including the <strong>1000 Genomes Project</strong>. As a consequence of this, data is organised in collections to reflect these different projects. Each of the directories in this directory contains data for a given collection. <strong>Data from the 1000 Genomes projects can be found under 1000_genomes_project.</strong></p>
<h3 id="Collection-layout"><a href="#Collection-layout" class="headerlink" title="Collection layout"></a>Collection layout</h3><p>Within each collection directory, you will find information in README files, which describe the data and any processing which has been done.</p>
<p>In addition, index files provide a catalogue of the files available for that collection. The body of each index file is tab delimited. Index files also have a header section with lines starting with ##, providing information about the file, data and the columns. The column header is a line starting with a single # immediately before the body of the file. The sequence indices contain the locations of the sequence data in the European Nucleotide Archive (ENA).</p>
<p>The data directory in each collection directory is where the data is housed. Data is organised by population and then by sample, for example, 1000_genomes_project/data/YRI/NA19150/.</p>
<h3 id="Further-information-2"><a href="#Further-information-2" class="headerlink" title="Further information"></a>Further information</h3><p>If you are looking for a specific file, a list of all files on the site can be found in <span class="exturl" data-url="ZnRwOi8vZnRwLjEwMDBnZW5vbWVzLmViaS5hYy51ay92b2wxL2Z0cC9jdXJyZW50LnRyZWU=" title="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/current.tree">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/current.tree<i class="fa fa-external-link"></i></span>. Should you have any queries, please contact <span class="exturl" data-url="bWFpbHRvOmluZm9AMTAwMGdlbm9tZXMub3Jn" title="mailto:info@1000genomes.org">info@1000genomes.org<i class="fa fa-external-link"></i></span>.</p>
</blockquote>
<h2 id="1000G-2504-high-coverage"><a href="#1000G-2504-high-coverage" class="headerlink" title="1000G_2504_high_coverage"></a>1000G_2504_high_coverage</h2><p>这是1000基因组2504基因高覆盖数据集合，如下是他的说明文件</p>
<blockquote>
<h3 id="1000-Genomes-2504-phase-3-panel-sequenced-to-high-coverage"><a href="#1000-Genomes-2504-phase-3-panel-sequenced-to-high-coverage" class="headerlink" title="1000 Genomes 2504 phase 3 panel sequenced to high coverage"></a>1000 Genomes 2504 phase 3 panel sequenced to high coverage</h3><p>This README refers to 30x Illumina NovaSeq sequencing of 2504 samples from the 1000 Genomes project phase 3 sample set. These data were generated at the New York Genome Center with funds provided by NHGRI Grant 3UM1HG008901-03S1. Please email <span class="exturl" data-url="bWFpbHRvOnNlcnZpY2VAbnlnZW5vbWUub3Jn" title="mailto:service@nygenome.org">service@nygenome.org<i class="fa fa-external-link"></i></span> with questions or interest in undertaking collaborative analysis of this dataset. All cell lines were obtained from the Coriell Institute for Medical Research and were consented for full public release of genomic data. Please see Coriell (<span class="exturl" data-url="aHR0cHM6Ly93d3cuY29yaWVsbC5vcmc=" title="https://www.coriell.org">https://www.coriell.org<i class="fa fa-external-link"></i></span>) for more information about specific cell lines. The following cell lines/DNA samples were obtained from the NIGMS Human Genetic Cell Repository at the Coriell Institute for Medical Research: [NA06984, NA06985, NA06986, NA06989, NA06994, NA07000, NA07037, NA07048, NA07051, NA07056, NA07347, NA07357, NA10847, NA10851, NA11829, NA11830, NA11831, NA11832, NA11840, NA11843, NA11881, NA11892, NA11893, NA11894, NA11918, NA11919, NA11920, NA11930. NA11931, NA11932, NA11933,<br>NA11992, NA11994, NA11995, NA12003, NA12004, NA12005, NA12006, NA12043, NA12044, NA12045, NA12046, NA12058, NA12144, NA12154, NA12155, NA12156, NA12234, NA12249, NA12272, NA12273, NA12275, NA12282, NA12283, NA12286, NA12287, NA12340, NA12341, NA12342, NA12347, NA12348, NA12383, NA12399, NA12400, NA12413,, NA12414, NA12489, NA12546, NA12716, NA12717, NA12718, NA12748, NA12749, NA12750, NA12751, NA12760, NA12761, NA12762, NA12763, NA12775, NA12776, NA12777, NA12778, NA12812, NA12813, NA12814, NA12815, NA12827, NA12828, NA12829, NA12830, NA12842, NA12843, NA12872, NA12873, NA12874, NA12878, NA12889, NA12890].</p>
<p>The sequence data is available in <span class="exturl" data-url="aHR0cHM6Ly93d3cuZWJpLmFjLnVrL2VuYS9kYXRhL3ZpZXcvUFJKRUIzMTczNg==" title="https://www.ebi.ac.uk/ena/data/view/PRJEB31736">ENA<i class="fa fa-external-link"></i></span> and a listing of files, with metadata, is available in the accompanyiing index.</p>
<h3 id="Analysis"><a href="#Analysis" class="headerlink" title="Analysis"></a>Analysis</h3><p>Analysis work is being done by a number of groups, working toward variant calling, including identification of structural variation.</p>
<p>Initial analysis has been done by NYGC, including aligning the data to GRCh38, creating the CRAMs in ENA. The document <span class="exturl" data-url="aHR0cDovL2Z0cC4xMDAwZ2Vub21lcy5lYmkuYWMudWsvdm9sMS9mdHAvZGF0YV9jb2xsZWN0aW9ucy8xMDAwR18yNTA0X2hpZ2hfY292L05ZR0NfYjM4X3BpcGVsaW5lX2Rlc2NyaXB0aW9uLnBkZg==" title="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_cov/NYGC_b38_pipeline_description.pdf">NYGC_b38_pipeline_description.pdf<i class="fa fa-external-link"></i></span> contains a description of that analysis work and details of the alignment pipeline.</p>
<p>Should you have questions about this data please contact <span class="exturl" data-url="bWFpbHRvOmluZm9AMTAwMGdlbm9tZXMub3Jn" title="mailto:info@1000genomes.org">info@1000genomes.org<i class="fa fa-external-link"></i></span></p>
</blockquote>
<h2 id="1000-genomes-project"><a href="#1000-genomes-project" class="headerlink" title="1000_genomes_project"></a>1000_genomes_project</h2><p>这是1000基因组的原始数据存储文件</p>
<blockquote>
<p>This directory contains sequence data generated by the 1000 Genomes Project, and alignment data of the reads to GRCh38.<br>Subsequent analysis results such as variant call sets will be in this directory when they become available. </p>
<h3 id="Sequence-Data"><a href="#Sequence-Data" class="headerlink" title="Sequence Data"></a>Sequence Data</h3><p>Moving forward we are aligning the raw sequence data unfiltered, as such we no long rehost the fastq files but instead our sequence index files point you to the FTP url for the fastq files on ENA servers</p>
<p>The data in the sequence index are categorised into 3 analysis groups (column 26) based on the library strategy used:</p>
<p>low coverage - Low coverage whole genome sequencing<br>exome - Whole exome sequencing<br>high coverage - PCR-free high coverage whole genome sequencing</p>
<p>When align to GRCh38,  data from the three analysis groups were aligned independently, producing three sets of CRAM files.</p>
<p>Some of the runs have a withdrawn flag (column 21); these runs were not included in subsequent analysis.<br>Column 22 indicates the reason why a run is withdrawn.  Below are the possible reasons:</p>
<p>SUPPRESSED IN ARCHIVE - runs were withdrawn by submitters so the fastq file is no longer available from ENA/SRA<br>TOO_SHORT - alll reads in a  whole genome sequencing run are shorter than 70bp or in a exome run shorter than 68bp<br>NOT_ILLUMINA - a run generated on a platform that is not Illumina.  For data consistency, we only include data produced on the Illumina platform for subsequent analysis. </p>
<h3 id="Alignment-Data"><a href="#Alignment-Data" class="headerlink" title="Alignment Data"></a>Alignment Data</h3><p>The sequence data described by the sequence index file (1000genomes.sequence.index) were aligned to GRCh38 using ALT-aware bwa-mem. Please see details of the alignment process in README.1000genomes.GRCh38DH.alignment.  Sample level CRAMs were produced; the CRAM files and ancillary files are listed in alignment index files in this directory.</p>
</blockquote>
<h1 id="总结"><a href="#总结" class="headerlink" title="总结"></a>总结</h1><p>1000基因组的数据文件非常大，存储达到1000G左右，所以存储是一个不小的问题，需要花费很多计算资源和数据存储资源，并且算法上面也是一个很高维度的计算方法才能对基因的问题有一个比较好的描述和解释，目前所采用的方法大多是传统的方法，就是因为目前的方法都是基于简单计算的，有很好的公式进行表示计算，但是人工智能领域的方法并没有很好的描述这个问题，现在的技术方法就是需要用人工智能的方法拟合传统公式所需要表达的思想，然后通过神经网络拟合的网络进行计算，得到更加精确的解释，但是这样也有一定的风险，因为人工智能的方法能不能刻画这个模型，或者刻画这个模型以后是不是有更好的精确解释，这是一个比较难表示的问题。</p>

      
    </div>

    
      


    

    
    
    

    

    
      
    
    

    
      <div>
        




  



<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>塘朗爬坡王</li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    
    <a href="http://univeryinli.github.io/2019/07/18/1000-genomes/" title="1000 Genomes">http://univeryinli.github.io/2019/07/18/1000-genomes/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC8="><i class="fa fa-fw fa-creative-commons"></i>BY-NC-SA</span> unless stating additionally.</li>
</ul>

      </div>
    

    <footer class="post-footer">
      
        
          
        
        <div class="post-tags">
          
            <a href="/tags/Gene常识/" rel="tag"><i class="fa fa-tag"></i> Gene常识</a>
          
        </div>
      

      
      
        <div class="post-widgets">
        

        

        
          
          <div class="social_share">
            
            
              <div id="needsharebutton-postbottom">
                <span class="btn">
                  <i class="fa fa-share-alt" aria-hidden="true"></i>
                </span>
              </div>
            
            
          </div>
        
        </div>
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2019/07/07/Ubuntu下搭建多用户多权限FTP/" rel="next" title="Ubuntu下搭建多用户多权限FTP">
                <i class="fa fa-chevron-left"></i> Ubuntu下搭建多用户多权限FTP
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/2019/07/19/problem/" rel="prev" title="google scholar vs baidu scholar-what is your problem">
                google scholar vs baidu scholar-what is your problem <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>


  </div>


          </div>
          

  
    <div class="comments" id="comments">
    </div>

  



        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            Table of Contents
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview-wrap">
            Overview
          </li>
        </ul>
      

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-overview">
          <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
            
              <img class="site-author-image" itemprop="image" src="https://s2.ax1x.com/2019/06/23/ZPsmxP.jpg" alt="塘朗爬坡王">
            
              <p class="site-author-name" itemprop="name">塘朗爬坡王</p>
              <div class="site-description motion-element" itemprop="description">NLP、Deep Leaning、Travel&Bicycle</div>
          </div>

          
            <nav class="site-state motion-element">
              
                <div class="site-state-item site-state-posts">
                
                  <a href="/archives/">
                
                    <span class="site-state-item-count">15</span>
                    <span class="site-state-item-name">posts</span>
                  </a>
                </div>
              

              
                
                
                <div class="site-state-item site-state-categories">
                  
                    
                      <a href="/categories/">
                    
                  
                    
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                    <span class="site-state-item-count">12</span>
                    <span class="site-state-item-name">categories</span>
                  </a>
                </div>
              

              
                
                
                <div class="site-state-item site-state-tags">
                  
                    
                      <a href="/tags/">
                    
                  
                    
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                    <span class="site-state-item-count">18</span>
                    <span class="site-state-item-name">tags</span>
                  </a>
                </div>
              
            </nav>
          

          

          

          
            <div class="links-of-author motion-element">
              
                <span class="links-of-author-item">
                  
                  
                    
                  
                  
                    
                  
                  <span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL3VuaXZlcnlpbmxp" title="GitHub &rarr; https://github.com/univeryinli"><i class="fa fa-fw fa-github"></i>GitHub</span>
                </span>
              
                <span class="links-of-author-item">
                  
                  
                    
                  
                  
                    
                  
                  <span class="exturl" data-url="bWFpbHRvOnVuaXZlcnlpbmxpQHFxLmNvbQ==" title="E-Mail &rarr; mailto:univeryinli@qq.com"><i class="fa fa-fw fa-envelope"></i>E-Mail</span>
                </span>
              
                <span class="links-of-author-item">
                  
                  
                    
                  
                  
                    
                  
                  <span class="exturl" data-url="aHR0cHM6Ly93d3cubGlua2VkaW4uY29tL2luL3VuaXZlcnlpbmxp" title="LinkedIn &rarr; https://www.linkedin.com/in/univeryinli"><i class="fa fa-fw fa-linkedin"></i>LinkedIn</span>
                </span>
              
            </div>
          

          
             <div class="cc-license motion-element" itemprop="license">
              
              
                
              
              
              
              <span class="exturl cc-opacity" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC8="><img src="/images/cc-by-nc-sa.svg" alt="Creative Commons"></span>
             </div>
          

          
          

          
            
          
          

        </div>
      </div>

      
      <!--noindex-->
        <div class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
            
            
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1000-Genomes-简介"><span class="nav-number">1.</span> <span class="nav-text">1000 Genomes 简介</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#基因组相关网址："><span class="nav-number">1.1.</span> <span class="nav-text">基因组相关网址：</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#基因组计划详细说明"><span class="nav-number">1.2.</span> <span class="nav-text">基因组计划详细说明</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#1000-Genomes-数据库结构-ftp"><span class="nav-number">2.</span> <span class="nav-text">1000 Genomes 数据库结构-ftp</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#How-to-download-files-using-Aspera"><span class="nav-number">2.1.</span> <span class="nav-text">How to download files using Aspera</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Download-Aspera"><span class="nav-number">2.1.1.</span> <span class="nav-text">Download Aspera</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Browser"><span class="nav-number">2.1.2.</span> <span class="nav-text">Browser</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Command-line"><span class="nav-number">2.1.3.</span> <span class="nav-text">Command line</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Further-details"><span class="nav-number">2.1.4.</span> <span class="nav-text">Further details</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#File-formats-and-descriptions"><span class="nav-number">2.2.</span> <span class="nav-text">File formats and descriptions</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#CRAM"><span class="nav-number">2.2.1.</span> <span class="nav-text">CRAM</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#CRAI"><span class="nav-number">2.2.2.</span> <span class="nav-text">CRAI</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#BAS"><span class="nav-number">2.2.3.</span> <span class="nav-text">BAS</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#INDEX"><span class="nav-number">2.2.4.</span> <span class="nav-text">INDEX</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Further-information"><span class="nav-number">2.2.5.</span> <span class="nav-text">Further information</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Structure-of-the-FTP-site"><span class="nav-number">2.3.</span> <span class="nav-text">Structure of the FTP site</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Top-level-of-the-site"><span class="nav-number">2.3.1.</span> <span class="nav-text">Top level of the site</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Directories"><span class="nav-number">2.3.2.</span> <span class="nav-text">Directories</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#changelog-details"><span class="nav-number">2.3.2.1.</span> <span class="nav-text">changelog_details</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#data"><span class="nav-number">2.3.2.2.</span> <span class="nav-text">data</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#data-collections"><span class="nav-number">2.3.2.3.</span> <span class="nav-text">data_collections</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#historical-data"><span class="nav-number">2.3.2.4.</span> <span class="nav-text">historical_data</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#phase1"><span class="nav-number">2.3.2.5.</span> <span class="nav-text">phase1</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#phase3"><span class="nav-number">2.3.2.6.</span> <span class="nav-text">phase3</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#pilot-data"><span class="nav-number">2.3.2.7.</span> <span class="nav-text">pilot_data</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#release"><span class="nav-number">2.3.2.8.</span> <span class="nav-text">release</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#technical"><span class="nav-number">2.3.2.9.</span> <span class="nav-text">technical</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Further-information-1"><span class="nav-number">2.3.3.</span> <span class="nav-text">Further information</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Populations"><span class="nav-number">2.4.</span> <span class="nav-text">Populations</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Populations-and-codes"><span class="nav-number">2.4.1.</span> <span class="nav-text">Populations and codes</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#IGSR-CRAM-Tutorial"><span class="nav-number">2.5.</span> <span class="nav-text">IGSR CRAM Tutorial</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Using-CRAM-files"><span class="nav-number">2.5.1.</span> <span class="nav-text">Using CRAM files</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#The-CRAM-reference-registry"><span class="nav-number">2.5.2.</span> <span class="nav-text">The CRAM reference registry</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#How-did-we-compress-the-BAMs-to-CRAMs"><span class="nav-number">2.5.3.</span> <span class="nav-text">How did we compress the BAMs to CRAMs</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#More-information-about-CRAM-format"><span class="nav-number">2.5.4.</span> <span class="nav-text">More information about CRAM format.</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#1000-Genomes-数据库结构-data-collections"><span class="nav-number">3.</span> <span class="nav-text">1000 Genomes 数据库结构-data_collections</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Data-collections"><span class="nav-number">3.1.</span> <span class="nav-text">Data collections</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Collections"><span class="nav-number">3.1.1.</span> <span class="nav-text">Collections</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Collection-layout"><span class="nav-number">3.1.2.</span> <span class="nav-text">Collection layout</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Further-information-2"><span class="nav-number">3.1.3.</span> <span class="nav-text">Further information</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1000G-2504-high-coverage"><span class="nav-number">3.2.</span> <span class="nav-text">1000G_2504_high_coverage</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1000-Genomes-2504-phase-3-panel-sequenced-to-high-coverage"><span class="nav-number">3.2.1.</span> <span class="nav-text">1000 Genomes 2504 phase 3 panel sequenced to high coverage</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Analysis"><span class="nav-number">3.2.2.</span> <span class="nav-text">Analysis</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1000-genomes-project"><span class="nav-number">3.3.</span> <span class="nav-text">1000_genomes_project</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Sequence-Data"><span class="nav-number">3.3.1.</span> <span class="nav-text">Sequence Data</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Alignment-Data"><span class="nav-number">3.3.2.</span> <span class="nav-text">Alignment Data</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#总结"><span class="nav-number">4.</span> <span class="nav-text">总结</span></a></li></ol></div>
            

          </div>
        </div>
      <!--/noindex-->
      

      
        <div class="back-to-top">
          <i class="fa fa-arrow-up"></i>
          
            <span id="scrollpercent"><span>0</span>%</span>
          
        </div>
      

    </div>
  </aside>
  


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">&copy; 2018 – <span itemprop="copyrightYear">2019</span>
  <span class="with-love" id="animate">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">塘朗爬坡王</span>

  
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-area-chart"></i>
    </span>
    
    <span title="Symbols count total">79k</span>
  

  
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    
    <span title="Reading time total">29 mins.</span>
  
</div>










        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>

  
    <span class="post-meta-item-icon">
      <i class="fa fa-user"></i>
    </span>
    <span class="site-uv" title="Total Visitors">
      <span class="busuanzi-value" id="busuanzi_value_site_uv"></span> people visited our site.
    </span>
  

  
    <span class="post-meta-divider">|</span>
  

  
    <span class="post-meta-item-icon">
      <i class="fa fa-eye"></i>
    </span>
    <span class="site-pv" title="Total Views">
      Total visits <span class="busuanzi-value" id="busuanzi_value_site_pv"></span> times.
    </span>
  
</div>









        
      </div>
    </footer>

    

    
      <div id="needsharebutton-float">
        <span class="btn">
          <i class="fa fa-share-alt" aria-hidden="true"></i>
        </span>
      </div>
    

    

    
  </div>

  

<script>
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>





  



  






  



  
    
    
      
    
  
  <script color="0,0,255" opacity="0.5" zindex="-1" count="99" src="//cdn.jsdelivr.net/gh/theme-next/theme-next-canvas-nest@1/canvas-nest.min.js"></script>











  



  
  <script src="//cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>

  
  <script src="//cdn.jsdelivr.net/npm/fastclick@1/lib/fastclick.min.js"></script>

  
  <script src="//cdnjs.cloudflare.com/ajax/libs/jquery_lazyload/1.9.7/jquery.lazyload.min.js"></script>

  
  <script src="//cdn.jsdelivr.net/npm/velocity-animate@1/velocity.min.js"></script>

  
  <script src="//cdn.jsdelivr.net/npm/velocity-animate@1/velocity.ui.min.js"></script>

  
  <script src="//cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.6/jquery.fancybox.min.js"></script>

  
  <script src="//cdn.jsdelivr.net/gh/theme-next/theme-next-reading-progress@1/reading_progress.min.js"></script>


  


  <script src="/js/utils.js?v=7.2.0"></script>

  <script src="/js/motion.js?v=7.2.0"></script>



  
  


  <script src="/js/affix.js?v=7.2.0"></script>

  <script src="/js/schemes/pisces.js?v=7.2.0"></script>




  
  <script src="/js/scrollspy.js?v=7.2.0"></script>
<script src="/js/post-details.js?v=7.2.0"></script>



  


  <script src="/js/next-boot.js?v=7.2.0"></script>


  
  <script src="/js/js.cookie.js?v=7.2.0"></script>
  <script src="/js/scroll-cookie.js?v=7.2.0"></script>


  
  <script src="/js/exturl.js?v=7.2.0"></script>


  
  
  
    
  

  

  
  
  


  
  

  

<script src="//cdnjs.cloudflare.com/ajax/libs/valine/1.3.7/Valine.min.js"></script>

<script>
  var GUEST = ['nick', 'mail', 'link'];
  var guest = 'nick,mail,link';
  guest = guest.split(',').filter(function(item) {
    return GUEST.indexOf(item) > -1;
  });
  new Valine({
    el: '#comments',
    verify: false,
    notify: false,
    appId: 'gvzxDpcfMG00bkkl9bF8DjRV-gzGzoHsz',
    appKey: 'eNWU8dxhUST7JSzUeK3uCM3T',
    placeholder: 'Just go go',
    avatar: 'mm',
    meta: guest,
    pageSize: '10' || 10,
    visitor: true,
    lang: '' || 'zh-cn'
  });
</script>




  


  
  <script>
    // Popup Window;
    var isfetched = false;
    var isXml = true;
    // Search DB path;
    var search_path = "search.xml";
    if (search_path.length === 0) {
      search_path = "search.xml";
    } else if (/json$/i.test(search_path)) {
      isXml = false;
    }
    var path = "/" + search_path;
    // monitor main search box;

    var onPopupClose = function (e) {
      $('.popup').hide();
      $('#local-search-input').val('');
      $('.search-result-list').remove();
      $('#no-result').remove();
      $(".local-search-pop-overlay").remove();
      $('body').css('overflow', '');
    }

    function proceedsearch() {
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay"></div>')
        .css('overflow', 'hidden');
      $('.search-popup-overlay').click(onPopupClose);
      $('.popup').toggle();
      var $localSearchInput = $('#local-search-input');
      $localSearchInput.attr("autocapitalize", "none");
      $localSearchInput.attr("autocorrect", "off");
      $localSearchInput.focus();
    }

    // search function;
    var searchFunc = function(path, search_id, content_id) {
      'use strict';

      // start loading animation
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay">' +
          '<div id="search-loading-icon">' +
          '<i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>' +
          '</div>' +
          '</div>')
        .css('overflow', 'hidden');
      $("#search-loading-icon").css('margin', '20% auto 0 auto').css('text-align', 'center');

      

      $.ajax({
        url: path,
        dataType: isXml ? "xml" : "json",
        async: true,
        success: function(res) {
          // get the contents from search data
          isfetched = true;
          $('.popup').detach().appendTo('.header-inner');
          var datas = isXml ? $("entry", res).map(function() {
            return {
              title: $("title", this).text(),
              content: $("content",this).text(),
              url: $("url" , this).text()
            };
          }).get() : res;
          var input = document.getElementById(search_id);
          var resultContent = document.getElementById(content_id);
          var inputEventFunction = function() {
            var searchText = input.value.trim().toLowerCase();
            var keywords = searchText.split(/[\s\-]+/);
            if (keywords.length > 1) {
              keywords.push(searchText);
            }
            var resultItems = [];
            if (searchText.length > 0) {
              // perform local searching
              datas.forEach(function(data) {
                var isMatch = false;
                var hitCount = 0;
                var searchTextCount = 0;
                var title = data.title.trim();
                var titleInLowerCase = title.toLowerCase();
                var content = data.content.trim().replace(/<[^>]+>/g,"");
                
                var contentInLowerCase = content.toLowerCase();
                var articleUrl = decodeURIComponent(data.url).replace(/\/{2,}/g, '/');
                var indexOfTitle = [];
                var indexOfContent = [];
                // only match articles with not empty titles
                if(title != '') {
                  keywords.forEach(function(keyword) {
                    function getIndexByWord(word, text, caseSensitive) {
                      var wordLen = word.length;
                      if (wordLen === 0) {
                        return [];
                      }
                      var startPosition = 0, position = [], index = [];
                      if (!caseSensitive) {
                        text = text.toLowerCase();
                        word = word.toLowerCase();
                      }
                      while ((position = text.indexOf(word, startPosition)) > -1) {
                        index.push({position: position, word: word});
                        startPosition = position + wordLen;
                      }
                      return index;
                    }

                    indexOfTitle = indexOfTitle.concat(getIndexByWord(keyword, titleInLowerCase, false));
                    indexOfContent = indexOfContent.concat(getIndexByWord(keyword, contentInLowerCase, false));
                  });
                  if (indexOfTitle.length > 0 || indexOfContent.length > 0) {
                    isMatch = true;
                    hitCount = indexOfTitle.length + indexOfContent.length;
                  }
                }

                // show search results

                if (isMatch) {
                  // sort index by position of keyword

                  [indexOfTitle, indexOfContent].forEach(function (index) {
                    index.sort(function (itemLeft, itemRight) {
                      if (itemRight.position !== itemLeft.position) {
                        return itemRight.position - itemLeft.position;
                      } else {
                        return itemLeft.word.length - itemRight.word.length;
                      }
                    });
                  });

                  // merge hits into slices

                  function mergeIntoSlice(text, start, end, index) {
                    var item = index[index.length - 1];
                    var position = item.position;
                    var word = item.word;
                    var hits = [];
                    var searchTextCountInSlice = 0;
                    while (position + word.length <= end && index.length != 0) {
                      if (word === searchText) {
                        searchTextCountInSlice++;
                      }
                      hits.push({position: position, length: word.length});
                      var wordEnd = position + word.length;

                      // move to next position of hit

                      index.pop();
                      while (index.length != 0) {
                        item = index[index.length - 1];
                        position = item.position;
                        word = item.word;
                        if (wordEnd > position) {
                          index.pop();
                        } else {
                          break;
                        }
                      }
                    }
                    searchTextCount += searchTextCountInSlice;
                    return {
                      hits: hits,
                      start: start,
                      end: end,
                      searchTextCount: searchTextCountInSlice
                    };
                  }

                  var slicesOfTitle = [];
                  if (indexOfTitle.length != 0) {
                    slicesOfTitle.push(mergeIntoSlice(title, 0, title.length, indexOfTitle));
                  }

                  var slicesOfContent = [];
                  while (indexOfContent.length != 0) {
                    var item = indexOfContent[indexOfContent.length - 1];
                    var position = item.position;
                    var word = item.word;
                    // cut out 100 characters
                    var start = position - 20;
                    var end = position + 80;
                    if(start < 0){
                      start = 0;
                    }
                    if (end < position + word.length) {
                      end = position + word.length;
                    }
                    if(end > content.length){
                      end = content.length;
                    }
                    slicesOfContent.push(mergeIntoSlice(content, start, end, indexOfContent));
                  }

                  // sort slices in content by search text's count and hits' count

                  slicesOfContent.sort(function (sliceLeft, sliceRight) {
                    if (sliceLeft.searchTextCount !== sliceRight.searchTextCount) {
                      return sliceRight.searchTextCount - sliceLeft.searchTextCount;
                    } else if (sliceLeft.hits.length !== sliceRight.hits.length) {
                      return sliceRight.hits.length - sliceLeft.hits.length;
                    } else {
                      return sliceLeft.start - sliceRight.start;
                    }
                  });

                  // select top N slices in content

                  var upperBound = parseInt('1');
                  if (upperBound >= 0) {
                    slicesOfContent = slicesOfContent.slice(0, upperBound);
                  }

                  // highlight title and content

                  function highlightKeyword(text, slice) {
                    var result = '';
                    var prevEnd = slice.start;
                    slice.hits.forEach(function (hit) {
                      result += text.substring(prevEnd, hit.position);
                      var end = hit.position + hit.length;
                      result += '<b class="search-keyword">' + text.substring(hit.position, end) + '</b>';
                      prevEnd = end;
                    });
                    result += text.substring(prevEnd, slice.end);
                    return result;
                  }

                  var resultItem = '';

                  if (slicesOfTitle.length != 0) {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + highlightKeyword(title, slicesOfTitle[0]) + "</a>";
                  } else {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + title + "</a>";
                  }

                  slicesOfContent.forEach(function (slice) {
                    resultItem += "<a href='" + articleUrl + "'>" +
                      "<p class=\"search-result\">" + highlightKeyword(content, slice) +
                      "...</p>" + "</a>";
                  });

                  resultItem += "</li>";
                  resultItems.push({
                    item: resultItem,
                    searchTextCount: searchTextCount,
                    hitCount: hitCount,
                    id: resultItems.length
                  });
                }
              })
            };
            if (keywords.length === 1 && keywords[0] === "") {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-search fa-5x"></i></div>'
            } else if (resultItems.length === 0) {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-frown-o fa-5x"></i></div>'
            } else {
              resultItems.sort(function (resultLeft, resultRight) {
                if (resultLeft.searchTextCount !== resultRight.searchTextCount) {
                  return resultRight.searchTextCount - resultLeft.searchTextCount;
                } else if (resultLeft.hitCount !== resultRight.hitCount) {
                  return resultRight.hitCount - resultLeft.hitCount;
                } else {
                  return resultRight.id - resultLeft.id;
                }
              });
              var searchResultList = '<ul class=\"search-result-list\">';
              resultItems.forEach(function (result) {
                searchResultList += result.item;
              })
              searchResultList += "</ul>";
              resultContent.innerHTML = searchResultList;
            }
          }

          if ('auto' === 'auto') {
            input.addEventListener('input', inputEventFunction);
          } else {
            $('.search-icon').click(inputEventFunction);
            input.addEventListener('keypress', function (event) {
              if (event.keyCode === 13) {
                inputEventFunction();
              }
            });
          }

          // remove loading animation
          $(".local-search-pop-overlay").remove();
          $('body').css('overflow', '');

          proceedsearch();
        }
      });
    }

    // handle and trigger popup window;
    $('.popup-trigger').click(function(e) {
      e.stopPropagation();
      if (isfetched === false) {
        searchFunc(path, 'local-search-input', 'local-search-result');
      } else {
        proceedsearch();
      };
    });

    $('.popup-btn-close').click(onPopupClose);
    $('.popup').click(function(e){
      e.stopPropagation();
    });
    $(document).on('keyup', function (event) {
      var shouldDismissSearchPopup = event.which === 27 &&
        $('.search-popup').is(':visible');
      if (shouldDismissSearchPopup) {
        onPopupClose();
      }
    });
  </script>





  

  

  
  

  
  

  


  
<script>
if ($('body').find('div.pdf').length) {
  $.ajax({
    type: 'GET',
    url: '//cdn.jsdelivr.net/npm/pdfobject@2/pdfobject.min.js',
    dataType: 'script',
    cache: true,
    success: function() {
      $('body').find('div.pdf').each(function(i, o) {
        PDFObject.embed($(o).attr('target'), $(o), {
          pdfOpenParams: {
            navpanes: 0,
            toolbar: 0,
            statusbar: 0,
            pagemode: 'thumbs',
            view: 'FitH'
          },
          PDFJS_URL: '/lib/pdf/web/viewer.html',
          height: $(o).attr('height') || '500px'
        });
      });
    },
  });
}
</script>


  

  
  <script>
    (function(){
      var bp = document.createElement('script');
      var curProtocol = window.location.protocol.split(':')[0];
      bp.src = (curProtocol === 'https') ? 'https://zz.bdstatic.com/linksubmit/push.js' : 'http://push.zhanzhang.baidu.com/push.js';
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(bp, s);
    })();
  </script>


  

  
  
  
    
  
  <script src="//cdn.jsdelivr.net/gh/theme-next/theme-next-needmoreshare2@1/needsharebutton.min.js"></script>
  <script>
    
      pbOptions = {};
      
        pbOptions.iconStyle = "box";
      
        pbOptions.boxForm = "horizontal";
      
        pbOptions.position = "bottomCenter";
      
        pbOptions.networks = "Weibo,Wechat,Douban,QQZone,Twitter,Facebook";
      
      new needShareButton('#needsharebutton-postbottom', pbOptions);
    
    
      flOptions = {};
      
        flOptions.iconStyle = "box";
      
        flOptions.boxForm = "horizontal";
      
        flOptions.position = "middleRight";
      
        flOptions.networks = "Weibo,Wechat,Douban,QQZone,Twitter,Facebook";
      
      new needShareButton('#needsharebutton-float', flOptions);
    
  </script>


  

  

  

  
<script>
  $('.highlight').not('.gist .highlight').each(function(i, e) {
    var $wrap = $('<div>').addClass('highlight-wrap');
    $(e).after($wrap);
    $wrap.append($('<button>').addClass('copy-btn').append('Copy').on('click', function(e) {
      var code = $(this).parent().find('.code').find('.line').map(function(i, e) {
        return $(e).text();
      }).toArray().join('\n');
      var ta = document.createElement('textarea');
      var yPosition = window.pageYOffset || document.documentElement.scrollTop;
      ta.style.top = yPosition + 'px'; // Prevent page scroll
      ta.style.position = 'absolute';
      ta.style.opacity = '0';
      ta.readOnly = true;
      ta.value = code;
      document.body.appendChild(ta);
      const selection = document.getSelection();
      const selected = selection.rangeCount > 0 ? selection.getRangeAt(0) : false;
      ta.select();
      ta.setSelectionRange(0, code.length);
      ta.readOnly = false;
      var result = document.execCommand('copy');
      
        if (result) $(this).text('Copied');
        else $(this).text('Copy failed');
      
      ta.blur(); // For iOS
      $(this).blur();
      if (selected) {
        selection.removeAllRanges();
        selection.addRange(selected);
      }
    })).on('mouseleave', function(e) {
      var $b = $(this).find('.copy-btn');
      setTimeout(function() {
        $b.text('Copy');
      }, 300);
    }).append(e);
  })
</script>


  

  

</body>
</html>
